src/repr/src/timestamp.rs

```diff
@@ -79,6 +80,15 @@ pub trait TimestampManipulation:
 
     /// Return the maximum value for this timestamp.
     fn maximum() -> Self;
+
+    /// Rounds up the timestamp to the time of the next refresh according to the given schedule.
+    /// Returns None if there is no next refresh.
+    fn round_up(&self, schedule: &RefreshSchedule) -> Option<Self>;
+
+    /// Rounds down `timestamp - 1` to the time of the previous refresh according to the given
+    /// schedule.
+    /// Returns None if there is no previous refresh.
+    fn round_down_m1(&self, schedule: &RefreshSchedule) -> Option<Self>;
```

Comment:
Missed this yesterday, let's use the full name. I didn't understand what `m1` means.

```suggestion
    fn round_down_minus_1(&self, schedule: &RefreshSchedule) -> Option<Self>;
```

Reply:
Renamed